mm2src/coins/lightning.rs

```diff
@@ -117,9 +124,14 @@ impl LightningCoin {
     #[inline]
     fn my_node_id(&self) -> String { self.channel_manager.get_our_node_id().to_string() }
 
-    fn get_balance_msat(&self) -> (u64, u64) {
-        self.channel_manager
-            .list_channels()
+    async fn list_channels(&self) -> Vec<ChannelDetails> {
+        let channel_manager = self.channel_manager.clone();
+        async_blocking(move || channel_manager.list_channels()).await
```

Comment:
This is all due to the `channel_state` Mutex
https://github.com/lightningdevkit/rust-lightning/blob/00f08c910ff87fecc580750df33d3ebf5ac1fceb/lightning/src/ln/channelmanager.rs#L702

Since `list_channels` uses `list_channels_with_filter` which waits to acquire the lock here https://github.com/lightningdevkit/rust-lightning/blob/00f08c910ff87fecc580750df33d3ebf5ac1fceb/lightning/src/ln/channelmanager.rs#L1743

And I believe, this can take some time, as this lock is locked everywhere when dealing with channels in rust-lightning, so it should require using `async_blocking` here. An example for a function that acquires the lock for sometime is `internal_funding_signed`
https://github.com/lightningdevkit/rust-lightning/blob/00f08c910ff87fecc580750df33d3ebf5ac1fceb/lightning/src/ln/channelmanager.rs#L4668 https://github.com/lightningdevkit/rust-lightning/blob/00f08c910ff87fecc580750df33d3ebf5ac1fceb/lightning/src/ln/channelmanager.rs#L4679 https://github.com/lightningdevkit/rust-lightning/blob/00f08c910ff87fecc580750df33d3ebf5ac1fceb/lightning/src/chain/chainmonitor.rs#L584
Where `persist_new_channel` saves the new monitor to the filesystem.

It's not always the case that `list_channels` will wait for an I/O process to complete to acquire the lock, I even think that this happens rarely, but the frequency with which this happens should increase with the number of channels open.